cpp/benchmarks/common/generate_benchmark_input.cpp

```diff
@@ -347,14 +347,22 @@ void copy_string(cudf::size_type src_idx,
 template <typename Char_gen>
 void append_string(Char_gen& char_gen, bool valid, uint32_t length, string_column_data& column_data)
 {
-  auto const idx = column_data.offsets.size() - 1;
-  column_data.offsets.push_back(column_data.offsets.back() + length);
-  std::generate_n(std::back_inserter(column_data.chars),
-                  column_data.offsets[idx + 1] - column_data.offsets[idx],
-                  [&]() { return char_gen(); });
-
-  // TODO: use empty string for invalid fields?
-  if (!valid) { cudf::clear_bit_unsafe(column_data.null_mask.data(), idx); }
+  if (!valid) {
+    auto const idx = column_data.offsets.size() - 1;
+    cudf::clear_bit_unsafe(column_data.null_mask.data(), idx);
+    // duplicate the offset value to indicate an empty row
+    column_data.offsets.push_back(column_data.offsets.back());
+    return;
+  }
+  std::generate_n(std::back_inserter(column_data.chars), length, [&]() mutable {
```

Comment:
Yes. I can change it to a for-loop if you want.

Reply:
Also, this change is not part of this PR but is from #7292

Reply:
I posted comments on that PR. Will resolve my comments here, re-review when that one is merged.

Reply:
@codereport what is the alternative here? We don't know the total number of characters in advance.

Reply:
@vuule in Haskell or C++20 it would be 
```
transform([](auto c) { return 1 char or 2 char; } | join;
```
however, it is a bit tricker in C++14. I am attending a C++ meetup - I will ask them :+1:

Reply:
I keep forgetting why I hate making PRs dependent on other PRs.

Reply:
@vuule General consensus was for loop would be better. However, comments about how type should support code points so they can be used nicely with algorithms. However, that seems out of scope for this PR.

Reply:
`transform([](auto c) { return 1 char or 2 char; } | join;`
wow, hard to believe that this is C++.

> type should support code points so they can be used nicely with algorithms

Interesting point. I noticed recently that overall in cuIO refactoring we would benefit from adding algorithm-friendly abstractions. Will be looking into this.

Reply:
> Interesting point. I noticed recently that overall in cuIO refactoring we would benefit from adding algorithm-friendly abstractions. Will be looking into this.

Note, `1 char or 2 char;` is pseudocode (just to be clear) :sweat_smile: